### PR TITLE
added warning for incompatibility with middleware_classes setting.

### PR DESCRIPTION
**Pull request by mohammadrabetian**
This pr is to resolve issue #1213

### CODE REVIEW

File: debug_toolbar/apps.py [hunk 1]
```diff
@@ -21,6 +21,17 @@ def check_middleware(app_configs, **kwargs):
     gzip_index = None
     debug_toolbar_indexes = []
 
+    # If old style MIDDLEWARE_CLASSES is being used, report an error.
+    if not settings.MIDDLEWARE:
+        errors.append(
+            Warning(
+                "debug_toolbar is incompatible with MIDDLEWARE_CLASSES setting.",
+                hint="Use MIDDLEWARE instead of MIDDLEWARE_CLASSES",
+                id="debug_toolbar.W000",
```
Comment (matthiask) [inline debug_toolbar/apps.py]: We count warnings from 1; 1-3 are already taken. This should be "W004".

File: debug_toolbar/apps.py [hunk 2]
```diff
@@ -21,6 +21,17 @@ def check_middleware(app_configs, **kwargs):
     gzip_index = None
     debug_toolbar_indexes = []
 
+    # If old style MIDDLEWARE_CLASSES is being used, report an error.
+    if not settings.MIDDLEWARE:
```
Comment (matthiask) [inline debug_toolbar/apps.py]: `django.conf.settings.is_overridden("MIDDLEWARE_CLASSES")` should work better for this.

**Comment by mohammadrabetian** [comment]
> Please add a test in `tests/test_integration.py` for this. Thanks!

it seems there's an integration test I've missed. I'll change my test.

**Comment by matthiask** [comment]
Nice! Now you're getting a test failure because Django 1.11 also raises warnings because of `MIDDLEWARE` / `MIDDLEWARE_CLASSES`

`[<Warning: level=30, msg="The MIDDLEWARE_CLASSES setting is deprecated in Django 1.10 and the MIDDLEWARE setting takes precedence. Since you've set MIDDLEWARE, the value of MIDDLEWARE_CLASSES is ignored.", hint=None, obj=None, id='1_10.W001'>,`

Maybe you want to unconditionally suppress this warning using `SILENCED_SYSTEM_CHECKS = ["1_10.W001"]  # Suppress the MIDDLEWARE_CLASSES warning of Django 1.11`

**Pull request merged**